server/routes/index.ts

```diff
@@ -745,20 +745,39 @@ export function defineRoutes(router: IRouter) {
   router.delete(
     {
       path: `${API_PREFIX}/configuration/cache`,
-      validate: false,
+      validate: {
+        body: schema.object({
+          dataSourceId: schema.maybe(schema.string()),
+        }),
+      },
     },
     async (context, request, response) => {
-      const client = context.security_plugin.esClient.asScoped(request);
-      let esResponse;
-      try {
-        esResponse = await client.callAsCurrentUser('opensearch_security.clearCache');
-        return response.ok({
-          body: {
-            message: esResponse.message,
-          },
-        });
-      } catch (error) {
-        return errorResponse(response, error);
+      if (!dataSourceEnabled || !request.body?.dataSourceId) {
```

Comment:
if dataSource is enabled and no datasource id is provided, should it still go in this block?

Reply:
Yes, if data source is enabled, there are 2 situations: 1. user makes a request to local cluster and local cluster doesn't have data source id since it is not a remote data source, in this case, we should still use the existing client; 2. user makes a request to remote data source, then we need to use the data source plugin from core provided client using the data source id. Hope it clarifies!